### PR TITLE
Improve docs and examples

### PR DESCRIPTION
**Pull request by igordejanovic**
In the recent couple of days I read through the docs and do some improvements along he way. 
Also I did a slight improvement in hello world and robot examples and tutorials based on a feedback I got from a user.

## Code review checklist

- [ ] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR)
- [ ] Title summarizes what is changing
- [ ] Commit messages are meaningful (see [this][commit messages] for details)
- [ ] Tests have been included and/or updated
- [ ] Docstrings have been included and/or updated, as appropriate
- [ ] Standalone docs have been updated accordingly
- [ ] Changelog(s) has/have been updated, as needed (see `CHANGELOG.md`, no need
      to update for typo fixes and such).

[commit messages]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: docs/rrel.md [hunk 1]
```diff
@@ -1,28 +1,47 @@
...
+rule. This is a link rule reference as it is enclosed inside of square brackets.
+It consists of three parts separated by `|`. The first part defines the target
+object type or its grammar rule. The second part defines what will parser match
+at the place of the reference. It would be a fully qualified name of the target
+object (thus `FQN`). The third part of the reference is RREL expression
+(`^packages*.classes`). Second and third part of the reference are optional. If
+second part is not given `ID` is assumed. If RREL expression is not given the
+default resolver, which search the reference in the global scope, will be used.
+
+Each reference in the model generally forms a dot separated name, which is
```
Comment (goto40) [inline docs/rrel.md]: generally --> by default [forms a dot separated name]

File: docs/rrel.md [hunk 2]
```diff
@@ -1,28 +1,47 @@
...
+matched by the second part of the link rule reference in the grammar, where a
+plain ID is just a special case. For example, a reference could be
+`package1.component4` or just `component4`. We could further generalize this by
+saying that a reference is a sequence of names where a plain ID is just a
+sequence of length 1. It doesn't have to be dot separated. A user could provide
+a custom match rule (like `FQN` in the above example) and a match processor to
+convert the matched string to a sequence of names. But for simplicity sake in
```
Comment (goto40) [inline docs/rrel.md]: There is also the possibility to define the separator sequence (by default a dot), as demonstrated in https://github.com/textX/textX/blob/master/tests/functional/test_scoping/test_rrel.py#L307 or below in sub section "RREL reference name deduction".

File: docs/rrel.md (hunk 1 above)
Comment (goto40) [reply docs/rrel.md]: (minor point - since it is mentioned below: you can also leave it as it is...)

File: docs/rrel.md (hunk 2 above)
Comment (goto40) [reply docs/rrel.md]: (same here; minor point - since it is mentioned below: you can also leave it as it is...)

**Review by goto40 — APPROVED** [review]
LGTM!
I commented one point - but realized the information is already provided below in the md file... I leave it up to you to decide if the information confuses the reader if provided to early or not...

File: docs/rrel.md (hunk 1 above)
Comment (igordejanovic) [reply docs/rrel.md]: Good catch. I agree, `by default` is better.

File: docs/rrel.md (hunk 2 above)
Comment (igordejanovic) [reply docs/rrel.md]: Sounds good. Fixed.

**Pull request merged**